### PR TITLE
[UR][CTS] Add urMemImageGetInfo tests and missing spec return values

### LINKED ISSUE #231: Adds CTS tests for `urMemImageGetInfo` entry-point
We should add conformance tests for `urMemImageGetInfo`.

Required Coverage:
* At least one successful (i.e. called with correct arguments and assert on expected result, not applicable to `CreateWithNativeHandle` entry points)
* If this a query entry-point or one which accepts flags etc : we should tests every possible enumeration (where possible).
* A test case for each returned error code (exclude disaster situations i.e. OUT_OF_MEMORY etc)

### PR DESCRIPTION
**Pull request by callumfare**
Resolves #231

### CODE REVIEW

File: test/conformance/memory/urMemImageGetInfo.cpp [hunk 1]
```diff
@@ -2,3 +2,83 @@
...
+TEST_P(urMemImageGetInfoTest, Success) {
+    ur_image_info_t info = getParam();
+    size_t size;
+    ASSERT_SUCCESS(urMemImageGetInfo(image, info, 0, nullptr, &size));
+    ASSERT_NE(size, 0);
+
+    if (const auto expected_size = image_info_size_map.find(info);
+        expected_size != image_info_size_map.end()) {
+        ASSERT_EQ(expected_size->second, size);
+    }
```
Comment (veselypeta) [inline test/conformance/memory/urMemImageGetInfo.cpp:37]:
```cpp
else {
    FAIL() << "<some error message>";
}
```

**Review by veselypeta — APPROVED** [review]
LGTM - just one suggestion

File: test/conformance/memory/urMemImageGetInfo.cpp [hunk 2]
```diff
@@ -2,3 +2,83 @@
...
+    size_t info_size = 0;
+    ASSERT_EQ_RESULT(
+        urMemImageGetInfo(image, UR_IMAGE_INFO_FORMAT, 0, &info_size, nullptr),
+        UR_RESULT_ERROR_INVALID_SIZE);
+}
+
+TEST_P(urMemImageGetInfoTest, InvalidSizeSmall) {
+    size_t info_size = 0;
+    ASSERT_EQ_RESULT(urMemImageGetInfo(image, UR_IMAGE_INFO_FORMAT,
+                                       sizeof(info_size) - 1, &info_size,
```
Comment (aarongreig) [inline test/conformance/memory/urMemImageGetInfo.cpp:68]: `UR_IMAGE_INFO_FORMAT` is an int so this wont always be too small to hold it

**Review by aarongreig — APPROVED** [review]

File: test/conformance/memory/urMemImageGetInfo.cpp (hunk 2 above)
Comment (callumfare) [reply test/conformance/memory/urMemImageGetInfo.cpp:70]: I thought `UR_IMAGE_INFO_FORMAT` was a struct with two enums (so presumably 64-bit total), but either way I've changed it to be int as there's no harm in making it smaller

Comment (aarongreig) [reply test/conformance/memory/urMemImageGetInfo.cpp:70]: you're right, got confused reading the header and thought it was another enum

**Pull request merged**